src/Tizen.Network.IoTConnectivity/Tizen.Network.IoTConnectivity/RemoteResource.cs

```diff
@@ -398,36 +403,39 @@ public async Task<RemoteResponse> GetAsync(ResourceQuery query = null)
             TaskCompletionSource<RemoteResponse> tcsRemoteResponse = new TaskCompletionSource<RemoteResponse>();
 
             IntPtr id = IntPtr.Zero;
-            lock (_responseCallbacksMap)
+            lock (_taskCompletionMap)
             {
-                id = (IntPtr)_responseCallbackId++;
+                id = (IntPtr)_responseCompletionId++;
             }
-            _responseCallbacksMap[id] = (IntPtr resource, int err, int requestType, IntPtr responseHandle, IntPtr userData) =>
-            {
-                IntPtr responseCallbackId = userData;
 
-                _responseCallbacksMap.Remove(responseCallbackId);
+            _taskCompletionMap[id] = tcsRemoteResponse;
+
+            _getResultCallback = (IntPtr resource, int err, int requestType, IntPtr responseHandle, IntPtr userData) =>
```

Comment:
It will be updated every method call, 
but it does not need you can make a static method and set on static constructor

``` c#
static void NativeResultCallbackHandler(IntPtr resource, int err, int requestType, IntPtr responseHandle, IntPtr userData) 
{
    IntPtr responseCompletionId = userData;
    TaskCompletionSource<RemoteResponse> responseCompletionSource = _taskCompletionMap[responseCompletionId];
    _taskCompletionMap.Remove(responseCompletionId);
....
}
private static Interop.IoTConnectivity.Client.RemoteResource.ResponseCallback _getResultCallback = NativeResultCallbackHandler;
```

Reply:
Thank you for your comment. I've fixed it.